### DISCUSSION TITLE
Migration from v1 to v2: SharedConfigState = session.SharedConfigEnable

### DISCUSSION THREAD

**Discussion by cyrilc-pro** [aws/aws-sdk-go-v2#2905] [General]
What is the proper way to rewrite the following v1 code with v2:
```
	options := session.Options{
		SharedConfigState: session.SharedConfigEnable,
	}
	return secretsmanager.New(session.Must(session.NewSessionWithOptions(options)))
```

I came up with:
```
	cfg, err := config.LoadDefaultConfig(ctx)
	if err != nil {
		return nil, err
	}
	return secretsmanager.NewFromConfig(cfg), nil
```

But I couldn't find a replacement for the `SharedConfigState` option in the documentation, nor in the code.

**Comment by Madrigal** [comment]
That's the right way to do it.

In v1, the session object only looked at `~/.aws/credentials`. From the [doc](https://docs.aws.amazon.com/sdk-for-go/api/aws/session/#SharedConfigState)

> By default NewSession will only load credentials from the shared credentials file (\~/.aws/credentials). If the AWS_SDK_LOAD_CONFIG environment variable is set to a truthy value the Session will be created from the configuration values from the shared config (\~/.aws/config) and shared credentials (\~/.aws/credentials) files. Using the NewSessionWithOptions with SharedConfigState set to SharedConfigEnable will create the session as if the AWS_SDK_LOAD_CONFIG environment variable was set. 

This has been simplified, and in v2, `config.LoadDefaultConfig(ctx)` automatically loads configuration from multiple sources, including shared config files (`~/.aws/config`) and shared credentials file (`~/.aws/credentials`). From the [docs](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/config#pkg-overview)

> The config package will load configuration from environment variables, AWS shared configuration file (\~/.aws/config), and AWS shared credentials file (\~/.aws/credentials).
Use the LoadDefaultConfig to load configuration from all the SDK's supported sources, and resolve credentials using the SDK's default credential chain.  

The behavior that was controlled by SharedConfigState in v1 is now the default behavior in v2. If you need to customize this behavior, v2 provides more granular control through functional options that you can pass to `config.LoadDefaultConfig()`
